libyul/AST.h

```diff
@@ -35,49 +35,67 @@ namespace solidity::yul
 
 using Type = YulString;
 
-struct TypedName { langutil::SourceLocation location; YulString name; Type type; };
+struct DebugData
+{
+	explicit DebugData(langutil::SourceLocation _location) : location(std::move(_location)) {}
+	langutil::SourceLocation location;
+};
+
+inline std::shared_ptr<DebugData const> createDebugData(langutil::SourceLocation _location = {})
+{
+	return std::make_shared<DebugData const>(_location);
```

Comment:
```suggestion
	return std::make_shared<DebugData const>(std::move(_location));
```

Reply:
I am not sure about using auto here. I think arguments of `make_shared` are rvalue references.
Also when I added move, clang-tidy was complaining that `_location` is copied for each invocation and is used as reference only.